theano/gpuarray/dnn.py

```diff
@@ -300,17 +302,42 @@ def __init__(self, border_mode, subsample=(1, 1), conv_mode='conv',
         self.border_mode = border_mode
         assert len(subsample) in (2, 3)
         self.subsample = subsample
+        if len(subsample) == 2:
+            subsample += (0,)
+        padding = [0, 0, 0]
+        bmode = 1
+        if isinstance(self.border_mode, tuple):
+            padding[0] = self.border_mode[0]
+            padding[1] = self.border_mode[1]
+            if len(self.border_mode) > 2:
+                padding[2] = self.border_mode[2]
+            bmode = 1
+        elif self.border_mode == "valid":
+            bmode = 1
+        elif self.border_mode == "half":
+            bmode = 2
+        elif self.border_mode == "full":
+            bmode = 0
+        else:
+            raise ValueError("Invalid value for border_mode")
         assert conv_mode in ('conv', 'cross')
         self.conv_mode = conv_mode
-
+        if self.conv_mode == 'conv':
+            conv_flag = 1  # 'CUDNN_CONVOLUTION'
+        elif self.conv_mode == 'cross':
+            conv_flag = 2  # 'CUDNN_CROSS_CORRELATION'
         assert precision in ['float16', 'float32', 'float64']
         self.precision = precision
+        precision_flag = 0
+        if self.precision == 'float16':
+            precision_flag = 1  # 'CUDNN_DATA_HALF'
+        elif self.precision == 'float32':
+            precision_flag = 2  # 'CUDNN_DATA_FLOAT'
+        else:
```

Comment:
@nouiz the else clause will accept an int value of 32? I don't think an else clause would make sense because we do an assert right before the if-else clause. So it can be only one of float16/32/64.
Another alternative would be precision = int(precision[5:]) and change the if-else statements to compare 16, 32 and 64

What do you think would be the best?